File: leetcode/242-Valid-Anagram/ValAnagram_001.java
public class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        
        int[] chars = new int[128];
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int idx = Character.getNumericValue(c);
            chars[idx] += 1;
        }
        
        for (int j = 0; j < t.length(); j++) {
            char c = t.charAt(j);
            int idx = Character.getNumericValue(c);
            if (chars[idx] == 0) {
                return false;
            }
            chars[idx] -= 1;
        }
        
        return true;
    }
}

File: oj/leetcode/242-Valid-Anagram/ValAnagram_001.py
class Solution:
    # @param {string} s
    # @param {string} t
    # @return {boolean}
    def anaRepresentation(self, s):
        p = {}
        for c in s:
            if c in p:
                p[c] += 1
            else:
                p[c] = 1
        return p
        
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        p = self.anaRepresentation(s)
        q = self.anaRepresentation(t)
        
        for c in p:
            if c not in q or (c in q and p[c] != q[c]):
                return False
        
        return True
